R/single_quotes_linter.R

```diff
@@ -2,20 +2,33 @@
 #' string constants.
 #' @export
 single_quotes_linter <- function(source_file) {
-  lapply(ids_with_token(source_file, "STR_CONST"),
+  if (is.null(source_file$full_parsed_content)) {
+    return(list())
+  }
+
+  content <- source_file$full_parsed_content
+  str_idx <- which(content$token == "STR_CONST")
+  squote_matches <- which(re_matches(
+    content[str_idx, 'text'],
+    rex(start, single_quote, any_non_double_quotes, single_quote, end)
+  ))
+
+  lapply(squote_matches,
     function(id) {
-      parsed <- with_id(source_file, id)
-      if (re_matches(parsed$text, rex(start, single_quote, any_non_double_quotes, single_quote, end))) {
+      with(content[str_idx[id], ], {
+        line <- source_file$file_lines[line1]
+        col2 <- if (line1 == line2) col2 else nchar(line)
         Lint(
           filename = source_file$filename,
-          line_number = parsed$line1,
-          column_number = parsed$col1,
+          line_number = line1,
+          column_number = col1,
           type = "style",
           message = "Only use double-quotes.",
-          line = source_file$lines[as.character(parsed$line1)],
-          ranges = list(sort(c(parsed$col1, parsed$col2))),
+          line = line,
+          ranges = list(c(col1, col2)),
```

Comment:
note the difference in logic here. instead of landing on the final column, land on either the final column, or the end of the first line, whichever comes first.

lmk if (1) this is as intended and/or (2) we should write a test for the difference.

Reply:
I think how you have it seems ok, but probably worth having a test case

Reply:
OK done. I adjusted one of the tests to include a check on the `ranges` output.